setup-env/action.yaml

```diff
@@ -71,11 +71,23 @@ runs:
           echo "Could not find existing node install, skipping node package installation"
         fi
 
+    - name: Determine pnpm version
+      if: env.PACKAGE_MANAGER == 'pnpm'
+      shell: bash
+      run: |
+        if [ -e package.json ] && command -v jq >/dev/null; then
+          PNPM_VERSION=$(jq -r '.packageManager|split("@")[1]' package.json || echo "")
```

Comment:
Splitting because the format in package.json is `pnpm@version-num` and we just need the version number.